src/index.ts

```diff
@@ -26,11 +26,10 @@ log += lottery.getDurations();
 log += '\n\n';
 log += `TOTAL DURATION ${lottery.totalDuration} IN MILLISECONDS.`
 
-if (!fs.existsSync(dir)){
-    fs.mkdirSync(dir);
-}
 const timestamp = new Date().toISOString();
-const fileName = `${dir}/lottery-results_${timestamp}.txt`;
-writeFile(fileName, log, () => console.log('Check result file ' + fileName));
+const fileName = `${filePrefix}${timestamp}`;
+const dataFormat = 'txt';
+const file = fileCreator.isDataType(fileName, dataFormat);
+fileCreator.writeToFile(file, log);
```

Comment:
Also, I think the public API of FileExporter should be only the writeToFile method (with the file type, as enum).
For me this interface method you created `isDataType(fileName, dataFormat)` should be something available in the private scope of the FileExporter. Let me know what you think.